### PR TITLE
timing-support

### PR DESCRIPTION
**Pull request by violetta-baeva**
Adding timing support. It will allow us to check how long each request takes to process.

### CODE REVIEW

File: include/haywire.h [hunk 1]
```diff
@@ -161,6 +162,7 @@ typedef struct
     void* headers;
     hw_string* body;
     int body_length;
+    clock_t begin_time;
```
Comment (nmdguerreiro) [inline include/haywire.h]: I would rename to `start_time`,

**Comment by nmdguerreiro** [comment]
Benchmark results (2x m4.xlarge machines, running with 4 server threads):

### this branch
```
[ec2-user@ip-172-31-5-244 wrk]$ ./wrk -c 50 -t 50 -d 10s -s pipelined_get.lua  --latency http://172.31.5.65:8000 -- 64
Running 10s test @ http://172.31.5.65:8000
  50 threads and 50 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency     2.63ms    1.92ms  45.18ms   72.67%
    Req/Sec    13.60k     3.79k   34.26k    71.63%
  Latency Distribution
     50%    2.61ms
     75%    4.03ms
     90%    6.82ms
     99%    0.00us
  6834322 requests in 10.10s, 1.12GB read
Requests/sec: 676703.33
Transfer/sec:    113.58MB
[ec2-user@ip-172-31-5-244 wrk]$ ./wrk -c 50 -t 50 -d 10s -s pipelined_get.lua  --latency http://172.31.5.65:8000 -- 64
unable to connect to 172.31.5.65:8000 Connection refused
[ec2-user@ip-172-31-5-244 wrk]$ ./wrk -c 50 -t 50 -d 10s -s pipelined_get.lua  --latency http://172.31.5.65:8000 -- 64
Running 10s test @ http://172.31.5.65:8000
  50 threads and 50 connections
```

### master
```
[ec2-user@ip-172-31-5-244 wrk]$ ./wrk -c 50 -t 50 -d 10s -s pipelined_get.lua  --latency http://172.31.5.65:8000 -- 64
Running 10s test @ http://172.31.5.65:8000
  50 threads and 50 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency     2.42ms    1.49ms  40.74ms   59.96%
    Req/Sec    14.47k     4.59k   49.79k    71.03%
  Latency Distribution
     50%    2.50ms
     75%    4.04ms
     90%    5.65ms
     99%    0.00us
  7266080 requests in 10.10s, 1.19GB read
Requests/sec: 719523.29
Transfer/sec:    120.77MB
```

**Comment by kellabyte** [comment]
Yeah, I was concerned getting wallclock every request or 2x per request would be expensive. That was my motivation to writing the HTTP partial response cache to avoid having to call the wallclock every request to get the date/time header value.

**Comment by nmdguerreiro** [comment]
This can be replaced by #96

**Comment by kellabyte** [comment]
Closing since #96 replaces this.

**Pull request closed without merging**